.github/workflows/update-durations.yml

```diff
@@ -0,0 +1,63 @@
+# This workflow is designed to run unit tests and open a pull request to update
+# JSON files that store test durations for better load balancing
+
+name: Update Durations
+on:
+  workflow_dispatch:
+  # Scheduled trigger every Saturday at 2:47am UTC
+  schedule:
+  - cron: '47 2 * * 6'
+
+concurrency:
+  group: update-durations-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  update-durations:
+    uses: ./.github/workflows/interface-unit-tests.yml
+    secrets:
+      codecov_token: ${{ secrets.CODECOV_TOKEN }}
+    with:
+      branch: master
+      upload_to_codecov: false
+      run_lightened_ci: true
+      skip_ci_test_jobs: 'torch-tests,autograd-tests,all-interfaces-tests,external-libraries-tests,qcut-tests,qchem-tests,gradients-tests,data-tests,device-tests'
+
+  merge-durations-files:
+    needs: update-durations
+    runs-on: ubuntu-latest
+    steps:
+      - name: Download artifacts
+        uses: actions/download-artifact@v4
+        with:
+          pattern: durations-*
+          path: ./
+          merge-multiple: false
+
+      - name: Merge artifacts into single file
+        run: |
+          mkdir durations
+          jq -s 'add' durations-core-*/*.json > durations/core_tests_durations.json
+          jq -s 'add' durations-jax-*/*.json > durations/jax_tests_durations.json
+          jq -s 'add' durations-tf-*/*.json > durations/tf_tests_durations.json
```

Comment:
A couple of silly questions from my end:
- Do we need mkdir -p in case we need to restart/rerun the workflow?
- Do we need to check for the existence of the json files before running each `jq` command (error handling)?

Reply:
> Do we need mkdir -p in case we need to restart/rerun the workflow?

No, that shouldn't be necessary. If the workflow needs to be re-run, the whole job would restart, spawning a new runner and reseting its state.

> Do we need to check for the existence of the json files before running each jq command (error handling)?

I don't think it should be necessary. The files will exist as long as:
* The jobs prior to this one that create the artifacts for the duration files finish successfully.
* The `download-artifact` action does not update with breaking changes that impact the file name/structure.

Reply:
Thanks, approving now!!